cmd/backfill-redis/main.go

```diff
@@ -0,0 +1,144 @@
+	"github.com/sigstore/rekor/pkg/types"
+
+	// these imports are to call the packages' init methods
+	_ "github.com/sigstore/rekor/pkg/types/alpine/v0.0.1"
+	_ "github.com/sigstore/rekor/pkg/types/cose/v0.0.1"
+	_ "github.com/sigstore/rekor/pkg/types/hashedrekord/v0.0.1"
+	_ "github.com/sigstore/rekor/pkg/types/helm/v0.0.1"
+	_ "github.com/sigstore/rekor/pkg/types/intoto/v0.0.1"
+	_ "github.com/sigstore/rekor/pkg/types/intoto/v0.0.2"
+	_ "github.com/sigstore/rekor/pkg/types/jar/v0.0.1"
+	_ "github.com/sigstore/rekor/pkg/types/rekord/v0.0.1"
+	_ "github.com/sigstore/rekor/pkg/types/rfc3161/v0.0.1"
+	_ "github.com/sigstore/rekor/pkg/types/rpm/v0.0.1"
+	_ "github.com/sigstore/rekor/pkg/types/tuf/v0.0.1"
+)
+
+var (
+	redisAddress = flag.String("address", "", "Address for Redis application")
+	redisPort    = flag.String("port", "", "Port to Redis application")
+	startIndex   = flag.Int("start", -1, "First index to backfill")
+	endIndex     = flag.Int("end", -1, "Last index to backfill")
+	rekorAddress = flag.String("rekor-address", "", "Address for Rekor, e.g. https://rekor.sigstore.dev")
+)
+
+func main() {
+	flag.Parse()
+
+	if *redisAddress == "" {
+		log.Fatal("address must be set")
+	}
+	if *redisPort == "" {
+		log.Fatal("port must be set")
+	}
+	if *startIndex == -1 {
+		log.Fatal("start must be set to >=0")
+	}
+	if *endIndex == -1 {
+		log.Fatal("end must be set to >=0")
+	}
+	if *rekorAddress == "" {
+		log.Fatal("rekor-address must be set")
+	}
+
+	cfg := radix.PoolConfig{}
+	redisClient, err := cfg.New(context.Background(), "tcp", fmt.Sprintf("%s:%s", *redisAddress, *redisPort))
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	rekorClient, err := client.GetRekorClient(*rekorAddress)
+	if err != nil {
+		log.Fatalf("creating rekor client: %v", err)
+	}
+
+	for i := *startIndex; i <= *endIndex; i++ {
+		params := entries.NewGetLogEntryByIndexParamsWithContext(context.Background())
+		params.SetLogIndex(int64(i))
+		resp, err := rekorClient.Entries.GetLogEntryByIndex(params)
+		if err != nil {
+			log.Fatalf("retrieving log uuid by index: %v", err)
+		}
+		success := true
+		for uuid, entry := range resp.Payload {
+			// uuid is the global UUID - tree ID and entry UUID
+			e, _, _, err := unmarshalEntryImpl(entry.Body.(string))
+			if err != nil {
+				fmt.Printf("error unmarshalling entry for %s: %v\n", uuid, err)
+				success = false
+				continue
+			}
+			keys, err := e.IndexKeys()
+			if err != nil {
+				fmt.Printf("error building index keys for %s: %v\n", uuid, err)
+				success = false
+				continue
+			}
+			for _, key := range keys {
+				if err := addToIndex(context.Background(), redisClient, key, uuid); err != nil {
+					success = false
+					fmt.Printf("error inserting UUID %s with key %s: %v\n", uuid, key, err)
+				}
+				fmt.Printf("Uploaded Redis entry %s, index %d, key %s\n", uuid, i, key)
+			}
+		}
+		if success {
+			fmt.Printf("Completed log index %d\n", i)
+		} else {
+			fmt.Printf("Errors with log index %d\n", i)
+		}
```

Comment:
nit: just log the index along with the error above instead of two separate lines?

Reply:
Cleaned up the error printing, now prints all errors per index